analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -412,27 +412,35 @@ public void Methods_Set_NotEmpty(bool condition)
     public void Method_Set_Empty(List<int> list, HashSet<int> set, Queue<int> queue, Stack<int> stack, ObservableCollection<int> obs, Dictionary<int, int> dict)
     {
         list.Clear();                   // Compliant
-        list.Clear();                   // FN
+        list.Clear();                   // Noncompliant
 
         set.Clear();                    // Compliant
-        set.Clear();                    // FN
+        set.Clear();                    // Noncompliant
 
         queue.Clear();                  // Compliant
-        queue.Clear();                  // FN
+        queue.Clear();                  // Noncompliant
 
         stack.Clear();                  // Compliant
-        stack.Clear();                  // FN
+        stack.Clear();                  // Noncompliant
 
         obs.Clear();                    // Compliant
-        obs.Clear();                    // FN
+        obs.Clear();                    // Noncompliant
 
         dict.Clear();                   // Compliant
-        dict.Clear();                   // FN
+        dict.Clear();                   // Noncompliant
 
         var empty = new List<int>();
         list.Add(5);
         list.Intersect(empty);          // Compliant
         list.Clear();                   // FN
+
+        list.Add(5);
+        list.RemoveAll(x => true);      // Compliant
+        list.Clear();                   // FN
```

Comment:
Is this FN due to `x=>true`?
Otherwise calling removeAll with some condition should not be considered as clearing the list.

On this note maybe add also a UT for this case?

```csharp
list.Add(5);
list.RemoveAll(x => x < 1);      // Compliant
list.Clear();      // Compliant
 ```